lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
-        for (i = 0; i < documentSubset[fieldName].length; i++) {
-          documentSubset[fieldName][i] = this.recurseApplyDefault(isUpdate, documentSubset[fieldName][i], collectionSpecSubset[fieldName].children);
+  /**
+   * @param {boolean} isUpdate
+   * @param {*} documentSubset
+   * @param {...StructuredFieldSpecification} collectionSpecSubset
+   */
+  recurseApplyDefault(isUpdate, documentSubset, collectionSpecSubset) {
+    Object.keys(collectionSpecSubset).forEach(fieldName => {
+      const
+        specSubset = collectionSpecSubset[fieldName],
+        field = documentSubset[fieldName];
+
+      if (documentSubset.hasOwnProperty(fieldName) && this.types[specSubset.type].allowChildren && specSubset.children) {
+        if (Array.isArray(field)) {
+          for (let i = 0; i < field.length; i++) {
+            field[i] = this.recurseApplyDefault(isUpdate, field[i], specSubset.children);
+          }
+        }
+        else {
+          documentSubset[fieldName] = this.recurseApplyDefault(isUpdate, field, specSubset.children);
         }
       }
-      else {
-        documentSubset[fieldName] = this.recurseApplyDefault(isUpdate, documentSubset[fieldName], collectionSpecSubset[fieldName].children);
+      else if (specSubset.defaultValue && (field === null || (!isUpdate && !documentSubset.hasOwnProperty(fieldName)))) {
+        documentSubset[fieldName] = specSubset.defaultValue;
       }
-    }
-    else if (collectionSpecSubset[fieldName].defaultValue && !isUpdate && (!documentSubset.hasOwnProperty(fieldName) || documentSubset[fieldName] === null)) {
-      documentSubset[fieldName] = collectionSpecSubset[fieldName].defaultValue;
-    }
-    else if (collectionSpecSubset[fieldName].defaultValue && isUpdate && documentSubset.hasOwnProperty(fieldName) && documentSubset[fieldName] === null) {
-      documentSubset[fieldName] = collectionSpecSubset[fieldName].defaultValue;
-    }
-  });
+    });
 
-  return documentSubset;
-};
+    return documentSubset;
+  }
 
-/**
- * @param {*} documentSubset
- * @param {StructuredFieldSpecification} collectionSpecSubset
- * @param {boolean} strictness
- * @param {string[]} errorMessages
- * @param {boolean} verbose
- */
-Validation.prototype.recurseFieldValidation = function validationRecurseFieldValidation (documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) {
-  if (strictness) {
-    if (!checkAllowedProperties(documentSubset, Object.keys(collectionSpecSubset))) {
+  /**
+   * @param {*} documentSubset
+   * @param {StructuredFieldSpecification} collectionSpecSubset
+   * @param {boolean} strictness
+   * @param {string[]} errorMessages
+   * @param {boolean} verbose
+   */
+  recurseFieldValidation(documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) {
+    if (strictness && !checkAllowedProperties(documentSubset, Object.keys(collectionSpecSubset))) {
       // We use a throw to be able to provide information about the field or the document in whole
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 98 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)